bundles/org.openhab.binding.generacmobilelink/src/main/java/org/openhab/binding/generacmobilelink/internal/handler/GeneracMobileLinkAccountHandler.java

```diff
@@ -61,191 +64,305 @@
-                }
-            });
-            HTTPResult result = futureResult.get();
-            logger.trace("Response - status: {} content: {}", result.responseCode, result.content);
-            return result;
-        } catch (ExecutionException e) {
-            return new HTTPResult(0, e.getMessage());
+
+            String parseSettings = matcher.group(1);
+            logger.debug("parseSettings: {}", parseSettings);
+            SignInConfig signInConfig = gson.fromJson(parseSettings, SignInConfig.class);
+
+            if (signInConfig == null) {
+                throw new IOException("Could not parse settings string");
+            }
+
+            Fields fields = new Fields();
+            fields.put("request_type", "RESPONSE");
+            fields.put("signInName", config.username);
+            fields.put("password", config.password);
+
+            Request selfAssertedRequest = httpClient.POST(LOGIN_BASE + "/SelfAsserted")
+                    .header("X-Csrf-Token", signInConfig.csrf).param("tx", "StateProperties=" + signInConfig.transId)
+                    .param("p", "B2C_1A_SignUpOrSigninOnline").content(new FormContentProvider(fields));
+
+            ContentResponse selfAssertedResposne = selfAssertedRequest.send();
+
+            logger.debug("selfAssertedRequest response {}", selfAssertedResposne.getStatus());
+
+            if (selfAssertedResposne.getStatus() != 200) {
+                throw new IOException("SelfAsserted: Bad response status: " + selfAssertedResposne.getStatus());
+            }
+
+            SelfAssertedResponse sa = gson.fromJson(selfAssertedResposne.getContentAsString(),
+                    SelfAssertedResponse.class);
+
+            if (sa == null) {
+                throw new IOException("SelfAsserted Could not parse response JSON");
+            }
+
+            if (!"200".equals(sa.status)) {
+                throw new InvalidCreendialsException("Invalid Credentials: " + sa.message);
+            }
+
+            Request confirmedRequest = httpClient.newRequest(LOGIN_BASE + "/api/CombinedSigninAndSignup/confirmed")
+                    .param("csrf_token", signInConfig.csrf).param("tx", "StateProperties=" + signInConfig.transId)
+                    .param("p", "B2C_1A_SignUpOrSigninOnline");
+
+            ContentResponse confirmedResponse = confirmedRequest.send();
+
+            if (confirmedResponse.getStatus() != 200) {
+                throw new IOException("CombinedSigninAndSignup bad response: " + confirmedResponse.getStatus());
+            }
+
+            String loginString = confirmedResponse.getContentAsString();
+            logger.trace("confirmedResponse: {}", loginString);
+            if (!submitPage(loginString)) {
+                throw new IOException("Error parsing HTML submit form");
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException | JsonSyntaxException e) {
+            throw new IOException(e);
```

Comment:
Consider if `InterruptedException` should be handled differently/separately, for example by  `Thread.currentThread().interrupt()` before throwing, or not catching it here.